### ISSUE TITLE
 looks like we got no XML document

### ISSUE THREAD

**Issue by tobby** [Garethp/php-ews#180]
looks like we got no XML document [/.../vendor/garethp/php-ews/src/API/NTLMSoapClient.php:118]

**Comment by reeperbahnause** [comment]
I was wrong. It has nothing to do with blocked contents.  
The email body I try to fetch, has an invalid xml character `&#xB;`   
This is why the response can not be parsed. 
I will try to figure out how to solve this problem.

**Comment by reeperbahnause** [comment]
Quick fix for now is the following but for sure there are better ways to do it

File `vendor/garethp/php-ews/src/API/NTLMSoapClient.php`

```php
    /**
     * Performs a SOAP request
     *
     * @link http://php.net/manual/en/function.soap-soapclient-dorequest.php
     *
     * @param string $request the xml soap request
     * @param string $location the url to request
     * @param string $action the soap action.
     * @param integer $version the soap version
     * @param integer $one_way
     * @return string the xml soap response.
     */
    public function __doRequest($request, $location, $action, $version, $one_way = 0)
    {
        $postOptions = array(
            'body' => $request,
            'headers' => array(
                'Connection' => 'Keep-Alive',
                'User-Agent' => 'PHP-SOAP-CURL',
                'Content-Type' => 'text/xml; charset=utf-8',
                'SOAPAction' => $action
            ),
            'verify' => $this->validate,
            'http_errors' => false
        );

        $postOptions = array_replace_recursive($postOptions, $this->auth);

        $response = $this->httpClient->post($location, $postOptions);

        $this->__last_request_headers = $postOptions['headers'];
        $this->_responseCode = $response->getStatusCode();

        //return $response->getBody()->__toString();

        $responseStr = $response->getBody()->__toString();
        $responseStr = str_replace('&#xB;', '', $responseStr);
        return $responseBody;
    }
```

**Comment by reeperbahnause** [comment]
I did some more research and I have a working solution.   
I found a solution using this StackOverflow answer:  
https://stackoverflow.com/a/8092672/6708205

Here is the code of the working solution:  

File `vendor/garethp/php-ews/src/API/NTLMSoapClient.php`

```php

    /**
     * Performs a SOAP request
     *
     * @link http://php.net/manual/en/function.soap-soapclient-dorequest.php
     *
     * @param string $request the xml soap request
     * @param string $location the url to request
     * @param string $action the soap action.
     * @param integer $version the soap version
     * @param integer $one_way
     * @return string the xml soap response.
     */
    public function __doRequest($request, $location, $action, $version, $one_way = 0)
    {
        $postOptions = array(
            'body' => $request,
            'headers' => array(
                'Connection' => 'Keep-Alive',
                'User-Agent' => 'PHP-SOAP-CURL',
                'Content-Type' => 'text/xml; charset=utf-8',
                'SOAPAction' => $action
            ),
            'verify' => $this->validate,
            'http_errors' => false
        );

        $postOptions = array_replace_recursive($postOptions, $this->auth);

        $response = $this->httpClient->post($location, $postOptions);

        $this->__last_request_headers = $postOptions['headers'];
        $this->_responseCode = $response->getStatusCode();

        // ======================================================
        // MODIFICATION START
        // ======================================================
        //return $response->getBody()->__toString();

        $responseStr = $response->getBody()->__toString();
        libxml_use_internal_errors(true);
        $dom = new \DOMDocument("1.0", "UTF-8");
        $dom->strictErrorChecking = false;
        $dom->validateOnParse = false;
        $dom->recover = true;
        $dom->loadXML($responseStr);
        $xml = simplexml_import_dom($dom);
        libxml_clear_errors();
        libxml_use_internal_errors(false);

        return $xml->asXML();
        // ======================================================
        // MODIFICATION END
        // ======================================================
    }
```

**Comment by Garethp** [comment]
That passes my unit tests. Do you want to submit a pull request, or would you prefer I just commit under my name and release?

**Comment by reeperbahnause** [comment]
Just commit it under your name. 
Thanks for your work!

**Comment by vytciv** [comment]
When we can expect this fix will be released? Cause I'm facing those invalid character issues quite for some time.

**Comment by Garethp** [comment]
This has been published as version `0.9.7`
